### PR TITLE
add location permission check for getCurrentWifiSSID and reScanAndLoa…

### PR DESCRIPTION
**Pull request by SpiriaJWF**
Fix : #257 , #265 , #258 and clarify #238

### CODE REVIEW

**Comment by eliaslecomte** [comment]
👏  Thanks for your contribution!

File: android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java [hunk 1]
```diff
@@ -576,4 +568,20 @@ private void stuffWifiConfigurationWithoutEncryption(final WifiConfiguration wif
     private String formatWithBackslashes(final String value) {
         return String.format("\"%s\"", value);
     }
+
+    private boolean isLocationPermissionGranted(final Promise promise) {
```
Comment (eliaslecomte) [inline android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java:572]: Not a 100% fan of the method name, as this method does the check _and_ rejects the promise, which isn't clear now.

Comment (eliaslecomte) [reply android/src/main/java/com/reactlibrary/rnwifi/RNWifiModule.java:572]: maybe `assertLocationPermissionGranted` or `rejectWhenLocationPermissionMissing`?

**Comment by elliottkember** [comment]
This patch looks good @eliaslecomte – are there any concerns about merging?

**Review by robrechtme — APPROVED** [review]

**Comment by robrechtme** [comment]
@JuanSeBestia looks good to me, can this be merged and published?

**Comment by JuanSeBestia** [comment]
I squash and rename to 

fix(Android): add location permission check for getCurrentWifiSSID and reScanAndLoadWifiList and change error message for clarity

because semantic commits is needed to release a new version

**Comment by eliaslecomte** [comment]
Thanks @JuanSeBestia for merging this & doing a release 🎉.

**Pull request merged**